### PR TITLE
Adds default stringify for api calls

### PR DESCRIPTION
**Pull request by mparke**
precursor to changes in #1191

### CODE REVIEW

File: src/api/apiActionReducerGenerator.js [hunk 1]
```diff
@@ -102,7 +105,13 @@ function genThunkPage(config, actions) {
     return async (dispatch, getState) => {
       const { token } = getState().authentication;
       const endpoint = `${config.endpoint(...ids, '')}?page=${page + 1}`;
-      const response = await fetch(token, endpoint, options);
+
+      const fetchOptions = _.merge({}, config.options, options);
+      if (fetchOptions.headers && fetchOptions.headers['X-Filter']) {
+        fetchOptions.headers['X-Filter'] = JSON.stringify(fetchOptions.headers['X-Filter']);
```
Comment (eatonphil) [inline src/api/apiActionReducerGenerator.js]: I see.

Comment (eatonphil) [inline src/api/apiActionReducerGenerator.js]: Just make the last two PRs one then?

File: src/api/apiActionReducerGenerator.js [hunk 2]
```diff
@@ -102,7 +105,13 @@ function genThunkPage(config, actions) {
     return async (dispatch, getState) => {
       const { token } = getState().authentication;
       const endpoint = `${config.endpoint(...ids, '')}?page=${page + 1}`;
-      const response = await fetch(token, endpoint, options);
+
+      const fetchOptions = _.merge({}, config.options, options);
```
Comment (eatonphil) [inline src/api/apiActionReducerGenerator.js]: This one is pretty easy to do without lodash: `{ ...config.options, ...options }`. (Unlike other things like groupBy)

Comment (mparke) [inline src/api/apiActionReducerGenerator.js]: my understanding was that spread will do a shallow merge/ compiles to Object.assign, when in the case of headers, we need a deep merge.

let me double check though

Comment (eatonphil) [inline src/api/apiActionReducerGenerator.js]: Ah. You're probably right

Comment (mparke) [inline src/api/apiActionReducerGenerator.js]: yeah, I think this is why:

[babel try it example](https://babeljs.io/repl/#?babili=false&evaluate=true&lineWrap=false&presets=es2015%2Cstage-2&targets=&browsers=&builtIns=false&code=function%20yes(state%3D'1')%20%7B%0A%20%20console.log(%2Bstate)%0A%7D%0A%0Alet%20obj%20%3D%20%7B%0A%20%20one%3A%201%2C%0A%20%20nested%3A%20%7B%0A%20%20%20%20two%3A%202%20%20%0A%20%20%7D%0A%7D%3B%0A%0Alet%20newObj%20%3D%20%7B%0A%20%20nested%3A%20%7B%0A%20%20%20%20three%3A%203%0A%20%20%7D%2C%0A%20%20...obj%0A%7D%3B%0A%0Aconsole.log(newObj)%3B)

**Pull request merged**